src/v/redpanda/admin_server.cc

```diff
@@ -5077,11 +5079,57 @@ admin_server::initialize_cluster_recovery(
   std::unique_ptr<ss::http::request> request,
   std::unique_ptr<ss::http::reply> reply) {
     reply->set_content_type("json");
+    if (need_redirect_to_leader(model::controller_ntp, _metadata_cache)) {
+        throw co_await redirect_to_leader(*request, model::controller_ntp);
+    }
+    auto& bucket_property = cloud_storage::configuration::get_bucket_config();
+    if (!bucket_property.is_overriden() || !bucket_property().has_value()) {
+        throw ss::httpd::bad_request_exception(
+          "Cluster recovery is not available. Missing bucket property");
+    }
+    cloud_storage_clients::bucket_name bucket(bucket_property().value());
+    auto result = ss::httpd::shadow_indexing_json::init_recovery_result{};
+    auto err = co_await _controller->get_cluster_recovery_manager().invoke_on(
+      cluster::cluster_recovery_manager::shard,
+      [bucket](auto& mgr) { return mgr.initialize_recovery(bucket); });
+    if (err == cluster::errc::not_leader_controller) {
+        throw co_await redirect_to_leader(*request, model::controller_ntp);
+    }
+    if (err == cluster::errc::cluster_already_exists) {
+        throw ss::httpd::base_exception{
+          "Recovery already active", ss::http::reply::status_type::conflict};
```

Comment:
Follow up with abhijat if there's someone who feels strongly about these codes

Reply:
Consider "precondition failed" or something similar (or something else)

Reply:
From the initial code walkthrough, there didn't seem to be consensus on changing these